xr/ipld/gen/gen.go

```diff
@@ -0,0 +1,100 @@
+//go:build ignore
+// +build ignore
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ipld/go-ipld-prime/schema"
+	gengo "github.com/ipld/go-ipld-prime/schema/gen/go"
+)
+
+func main() {
+	ts := schema.TypeSystem{}
+	ts.Init()
+	adjCfg := &gengo.AdjunctCfg{
+		CfgUnionMemlayout: map[schema.TypeName]string{
+			"Node_IPLD": "interface", // Use a more pointer-heavy memory layout for this type; this is necessary because it is recursive.
+		},
+	}
+
+	pkgName := "xr"
+
+	// This needs to preclude to have Type Kinds available.
+	ts.Accumulate(schema.SpawnString("String"))
+	// ts.Accumulate(schema.SpawnBytes("Blob"))
+	// ts.Accumulate(schema.SpawnInt("Int"))
+	// ts.Accumulate(schema.SpawnFloat("Float"))
+	// ts.Accumulate(schema.SpawnBool("Bool"))
+
+	ts.Accumulate(schema.SpawnString("String_IPLD"))
+	ts.Accumulate(schema.SpawnBytes("Blob_IPLD"))
+	ts.Accumulate(schema.SpawnInt("Int_IPLD"))
+	ts.Accumulate(schema.SpawnFloat("Float_IPLD"))
+	ts.Accumulate(schema.SpawnBool("Bool_IPLD"))
+
+	ts.Accumulate(schema.SpawnUnion("Node_IPLD",
+		[]schema.TypeName{ // Note that these are somewhat redundant statements due to reasons having to do with how we defined the schema DMT.  The DSL is not so redundant.
+			"String_IPLD",
+			"Blob_IPLD",
+			"Int_IPLD",
+			"Float_IPLD",
+			"Bool_IPLD",
+			"Dict_IPLD",
+			"Set_IPLD",
+		},
+
+		// Keys for the JSON representation of types
+		schema.SpawnUnionRepresentationKeyed(map[string]schema.TypeName{
+			"String": "String_IPLD",
+			"Blob":   "Blob_IPLD",
+			"Int":    "Int_IPLD",
+			"Float":  "Float_IPLD",
+			"Bool":   "Bool_IPLD",
+			"Dict":   "Dict_IPLD",
+			"Set":    "Set_IPLD",
+		}),
+	))
+
+	ts.Accumulate(schema.SpawnStruct("Dict_IPLD",
+		[]schema.StructField{
+			// Notice the lack of field called "type" -- that is expressed in the keyed union, which wraps this, instead: therefore it's not necessary to repeat here.
+			schema.SpawnStructField("Tag", "String", true, false),        // The bools here say "is optional; is not nullable".
```

Comment:
This string is not optional in our data model. Just: true -> false.

Reply:
You are right! I missed it from what Eric shared. Thanks!